### DISCUSSION TITLE
Why no apt-get package

### DISCUSSION THREAD

**Discussion by ghost** [debauchee/barrier#1273] [Q&A]
So what I found it how to install it on Debian:

https://snapcraft.io/install/barrier/debian#install

sudo apt update
sudo apt install snapd
sudo snap install core

sudo snap install barrier
```

apt-get install snapd
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  apparmor libapparmor-perl libpam-systemd snap-confine systemd systemd-shim ubuntu-core-launcher
Suggested packages:
  apparmor-profiles apparmor-profiles-extra apparmor-utils systemd-ui systemd-container policykit-1
The following NEW packages will be installed:
  apparmor libapparmor-perl libpam-systemd snap-confine snapd systemd systemd-shim ubuntu-core-launcher
0 upgraded, 8 newly installed, 0 to remove and 342 not upgraded.
Need to get 8760 kB of archives.
After this operation, 42.3 MB of additional disk space will be used.

```

My problem with this is that it would install ShitD, please no!

Can't you just provide a deb package?

Thank you.

**Comment by bheinks** [comment]
https://packages.debian.org/stable/source/barrier

**Comment by K0-RR** [comment]
What are you talking about? Barrier is in the Debian repository and if you don't like snap you can use the flatpak version.

**Comment by ghost** [comment]
SHIT1:

```
The following additional packages will be installed:
  apparmor libapparmor-perl libpam-systemd snap-confine systemd systemd-shim ubuntu-core-launcher


snap install core
error: cannot perform the following tasks:
- Mount snap "core" (11167) ([start snap-core-11167.mount] failed with exit status 1: Failed to start snap-core-11167.mount: Unknown unit: snap-core-11167.mount
See system logs and 'systemctl status snap-core-11167.mount' for details.
)

```
SHIT2:

```
apt install flatpak
flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
flatpak install flathub com.github.debauchee.barrier

```

```
[  113.361862] barriers[4528]: segfault at fb1 ip 0000000000000fb1 sp 00007ffc11e44ee8 error 14 in barriers[56431358d000+19000]
[  113.361876] Code: Bad RIP value.
[  114.378193] barriers[4548]: segfault at fb1 ip 0000000000000fb1 sp 00007fffd81ebd48 error 14 in barriers[564a1842f000+19000]
[  114.378201] Code: Bad RIP value.
[  115.414878] barriers[4568]: segfault at fb1 ip 0000000000000fb1 sp 00007ffe6fc40a38 error 14 in barriers[5580ea18e000+19000]
[  115.414889] Code: Bad RIP value.
[  116.433560] barriers[4588]: segfault at fb1 ip 0000000000000fb1 sp 00007fffb6c69398 error 14 in barriers[55e7b1ca6000+19000]
[  116.433565] Code: Bad RIP value.
[  117.471698] barriers[4608]: segfault at fb1 ip 0000000000000fb1 sp 00007ffcf5b6eb48 error 14 in barriers[55d71c950000+19000]
[  117.471712] Code: Bad RIP value.
[  118.508886] barriers[4628]: segfault at fb1 ip 0000000000000fb1 sp 00007ffc4fffeb98 error 14 in barriers[5607d8a91000+19000]
[  118.508901] Code: Bad RIP value.
[  119.528540] barriers[4648]: segfault at fb1 ip 0000000000000fb1 sp 00007ffd404dddb8 error 14 in barriers[557473638000+19000]
[  119.528545] Code: Bad RIP value.
[  120.542840] barriers[4668]: segfault at fb1 ip 0000000000000fb1 sp 00007ffc54977d58 error 14 in barriers[55c96cdab000+19000]
[  120.542846] Code: Bad RIP value.
[  121.561374] barriers[4688]: segfault at fb1 ip 0000000000000fb1 sp 00007ffdee745da8 error 14 in barriers[556ce2db3000+19000]
[  121.561379] Code: Bad RIP value.

```
Thank you very much for fucking up like 4 hours of my time with your asswipe product...

**Comment by liamgconnolly** [comment]
You should totally ask for a refund! what awful customer service!

Oh wait it's FOSS product...

Sorry there wasn't a big enough spoon in the known universe to feed you.

**Comment by K0-RR** [comment]
LOL, how did `flatpak install X` and `snap install X` wasted you 4 hours? Why didn't you just download the deb file like you wanted to?

**Comment by gnutechie** [comment]
> Thank you very much for fucking up like 4 hours of my time with your asswipe product...

Why on earth would you use SNAP packages on Debian? What do you think Debian is? Ubuntu?  Use the one that's already packaged in Debian.

sudo apt-get install barrier should do just fine... or if you would like to get a Debian package you need to ask more politely.

Manners Maketh Man

But you were asleep at that time I guess.

**Comment by ghost** [comment]
Cause Im using Debian 9 it's only in buster:

https://packages.debian.org/buster/barrier

Why would I upgrade my OS for this trash?
Yeah I found out it from other thread that if you dont have IPV6 support then it crashes like this so I had to recompile the kernel and reboot FUUUUUUUUUUUUUUUUUUUUUUUUUUU

**Comment by MatthewCroughan** [comment]
:cowboy_hat_face: Nix Cowboy here.

`nix-shell -p barrier` to get it temporarily in your shell. Once you exit that shell, it will no longer be available to execute on your `PATH`.

Or add it to your system config for all users to have access to barrier permanently like this:
```nix
{
   environment.systemPackages = with pkgs; [ barrier ];
}
```

I've made a commit to show you how easy this is [here](https://github.com/MatthewCroughan/nixcfg/commit/f4e0dc38a2346a6634ab45019644f205f75d190f)

**Comment by MatthewCroughan** [comment]
You can even build a whole virtual machine with `barrier` inside of it, just put the following in a file called `vm.nix` and run `nixos-shell` on it.

1. Put this into a file called `vm.nix`
```nix
{ pkgs, ... }: {
  environment.systemPackages = with pkgs; [ barrier ];

  users.users.matthew = {
    isNormalUser = true;
    extraGroups = [ "wheel" ]; # Enable ‘sudo’ for the user.
    password = "aptsux";
  };
}
```

2. Do this
```
nix-shell -p nixos-shell
nixos-shell vm.nix
```

You'll boot into a virtual machine wherein the user `matthew` has the password `aptsux` and has `barrier` on the `PATH` ready to execute.
 
# Demo
[![asciicast](https://asciinema.org/a/PXOsmdh3aYkultorHbNKN4pT6.svg)](https://asciinema.org/a/PXOsmdh3aYkultorHbNKN4pT6)

**Comment by shymega** [comment]
Maintainer here.

a) You're being disrespectful toward a *free product*, and the developers. Don't be so entitled.

b) There's already a Barrier package in Debian.

c) Stop using profanity.

We can resolve your issues, but only if you are not abusive.

If this issue deteriorates, I will have no hesitation in locking the issue.

To everyone else: thank you for being so patient and helping with this issue. My apologies for not turning up sooner... Nix is a good idea. Never used it myself though.

**Comment by gnutechie** [comment]
> Cause Im using Debian 9 it's only in buster:
> 
> https://packages.debian.org/buster/barrier
> 
> Why would I upgrade my OS for this trash?
> Yeah I found out it from other thread that if you dont have IPV6 support then it crashes like this so I had to recompile the kernel and reboot FUUUUUUUUUUUUUUUUUUUUUUUUUUU

Well perhaps in the future you can tell us first what version of Debian you're using. That would go a long way to getting an answer to your question.

And since it is in the main repository on Debian 10 that means that it's packaged for Debian 10 and tested for that particular Debian version.
 
Now if you would like the developer and/or the maintainer to try and give Debian 9 a go and try to make Barrier work with it you need to ask in a different way; since it means they have to go out of their way  and do all kinds of extra work (if they decide to do so) to accommodate a user that won't upgrade (for whatever reason) their Debian installation (for which they are not responsible for). 

You can try Flatpak since in Debian 9 that's probably the only way you could get it running for you... and since you don't want to recompile the kernel... and God Forbid reboot... 

To put out the fires from your F bombs without getting you banned in the process... here's what you can try:

Open a terminal and type:

1) sudo apt-get install flatpak
to install flatpak

2) sudo flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
To add the flathub repository so you can download all of the available packages including barrier

3) flatpak install flathub com.github.debauchee.barrier
to install barrier (execute this one as a normal user aka without sudo)

4) flatpak run com.github.debauchee.barrier
to run barrier

Then you can check whether barrier will run on Debian 9

So you see @feraldruid22, it's not just as simple as "Why no apt-get package?". 

Hope this helps you out.

**Comment by shymega** [comment]
Thank you for this calm response. Well laid out, and well said.

With regards to IPv6: you do *not* need to recompile the kernel. Nor reboot. Its a kernel parameter, or you could set it by creating a config in `/etc/sysctl.d`, which will take effect on reboot, A quick Google returns: https://www.techrepublic.com/article/how-to-disable-ipv6-on-linux/

It's not that difficult. If you don't want to reboot straight away, you should be able to reload sysctl, but off the top of my head, I cannot recall the specifics. I'm sure it's easily googleable.

With regards to supporting Debian 9, I don't intend to make any special effort to support it. Especially not after your behaviour. 

I'm not sure why you don't want to upgrade to Debian 10 - is there a reason? On the README, there is an embed of Barrier packages on various distros. See: https://repology.org/project/barrier/versions. That should give you a rough idea of what is supported on Debian, and elsewhere in terms of Barrier versions.

**Comment by shymega** [comment]
Closing - given the lack of reply from OP, and the fact we have all tried to assist as well, I see no reason to keep this open.

**Comment by clort81** [comment]
The responses were informative. I learned some things. Thanks.

**Comment by shymega** [comment]
This ticket is closed, please don't bump when unnecessary. Locking now.
